test/cpp/test_xla_sharding.cpp

```diff
@@ -239,5 +241,72 @@ TEST_F(XLAShardingTest, InputHandler) {
                                  at::kFloat));
 }
 
+using TensorShardingStore = ShardingUtil::TensorShardingStore;
+
+TEST_F(XLAShardingTest, TensorShardingStore) {
+  TensorShardingStore::ShardingInfo info{
+      std::make_shared<XLATensor::ShardingSpec>(
+          xla::HloSharding::Replicate().ToProto()),
+      torch::lazy::Value()};
+
+  int handle = 1;
+  TensorShardingStore::RegisterShardingInfo(handle, info);
+  EXPECT_TRUE(TensorShardingStore::GetShardingInfo(0) == nullptr);
+  EXPECT_TRUE(TensorShardingStore::GetShardingInfo(handle) != nullptr);
+
+  TensorShardingStore::UnregisterShardingInfo(handle);
+  EXPECT_TRUE(TensorShardingStore::GetShardingInfo(handle) == nullptr);
+
+  TensorShardingStore::RegisterShardingInfo(handle, info);
+  TensorShardingStore::Reset();
+  EXPECT_TRUE(TensorShardingStore::GetShardingInfo(handle) == nullptr);
+}
+
+TEST_F(XLAShardingTest, ShardInputDataNodes) {
+  if ((xla::sys_util::GetEnvString(xla::env::kEnvPjRtDevice, "") == "") ||
+      (xla::ComputationClient::Get()->GetLocalDevices().size() < 2)) {
+    GTEST_SKIP()
+        << "`PJRT_DEVICE` is not set, with more than 2 local devices, ("
+        << xla::ComputationClient::Get()->GetLocalDevices().size()
+        << " local devices detected).";
+  }
+
+  at::Tensor tensor = at::ones({8}, at::TensorOptions(at::kFloat));
+  const torch::lazy::BackendDevice* device = GetDefaultDevice();
+  torch::lazy::Value data_node = torch::lazy::MakeNode<DeviceData>(
+      std::move(TensorToXlaData(tensor, *device)));
+  EXPECT_TRUE(xla::ComputationClient::Get()
+                  ->GetDataShards(UnwrapXlaData(
+                      torch::lazy::getBackend()->GetComputationDataFromNode(
+                          data_node.node.get())))
+                  .size() == 1);
```

Comment:
can you add a comment here? Took me a while to figure out what this test is trying to check.